src/streamer.rs

```diff
@@ -468,6 +479,9 @@ fn recv_window(
             let p = b.write().expect("'b' write lock in fn recv_window");
             (p.get_index()?, p.meta.size)
         };
+        if pix > *consumed + WINDOW_SIZE {
+            continue;
```

Comment:
this would introduce a leak of the blob, need a recycler.recycle(b); before continue;

Reply:
it should be >= as well